### PR TITLE
projects: use query string

### PR DESCRIPTION
**Pull request by tharvik**
Avoid using JSON encoded params and use more classical way to query for data. It also avoid serializing empty values as to not bloat the URL too much.

Closes: #229

### CODE REVIEW

**Review by ineiti — COMMENTED** [review]
I thought we discussed to keep the `fragment` instead of changing it to a `query`?

**Comment by tharvik** [comment]
From what I remember, we discussed about it but didn't reach a conclusion.
* fragement are for reference inside a document, such as an heading
* query is, well, for queries, such as filtering or getting specific results out of a db
The latter seems more our case

**Comment by ineiti** [comment]
The URLs I linked to also discuss fragments that are not used to link to specific places in a document, see the wikipedia article.

So please use the fragment, and not the query.

**Comment by tharvik** [comment]
> The URLs I linked to also discuss fragments that are not used to link to specific places in a document, see the wikipedia article.

I still don't see it in the wikipedia article, they talk about using it for jumping to a section (a block of text, a range of line), but not for filtering a list or querying a db.

> So please use the fragment, and not the query.

Anyway, pushing a version which uses the fragment.

**Comment by tharvik** [comment]
> Will you do the rewrite of the URL from the 'old' style to the 'new' style in the ansible-PR?

I actually can't do it in the ansible-PR, as fragment aren't sent to server. So we might need to keep the JSON handling around for a very long time (though I don't think it impacts much people, so maybe we don't care about it?).

I'm pushing a commit handling that, your call if that's needed or not.

> With regard to the Wikipedia document:

Usage is not definition, it's not because we can use fragment to emulate a query behavior that we should. Anyway, I won't be arguing on it with GitHub, but be ready to talk about it on the next coffee :stuck_out_tongue_closed_eyes:

File: views/projects.tpl [hunk 1]
```diff
@@ -60,6 +60,12 @@
                 }
             })
 
+            // handle old JSON encoding of query
+            const old_json_matches = decodeURIComponent(window.location.hash)
+                .match(/^#{"dropdown":"([^"]*)","input":"([^"]*)"}$/);
```
Comment (ineiti) [inline views/projects.tpl:65]: Did you try if this works?

Comment (tharvik) [reply views/projects.tpl:65]: Yeah, of course, it doesn't on your side?

Comment (ineiti) [reply views/projects.tpl:65]: Sorry I doubted you...

**Review by ineiti — APPROVED** [review]

**Pull request merged**